proposals/NNNN-add-preferred-FPS.md

```diff
@@ -0,0 +1,128 @@
+This proposal allows head units to specify the preferred FPS, so that mobile library can refer to the specified value. 
+
+## Proposed solution
+
+The proposed solution is to add preferred FPS to VideoStreamingCapability. On a head unit, preferred FPS can be varied on screen resolution; i.e. frame rate can be higher for lower resolution, but the preferred FPS value can be used for the case where screen resolution is the preferred resolution value.
+Mobile Proxies should respect for the preferred FPS value, as well as they do for preferred resolution, maximum bitrate, etc.
+
+## Detailed design
+
+### RPC change (Mobile_API and HMI_API)
+
+Add preferredFPS to VideoStreamingCapability struct in both APIs.
+
+```xml
+    <struct name="VideoStreamingCapability" since="4.5">
+        <description>Contains information about this system's video streaming capabilities.</description>
+        ...
+        <!-- new param -->
+        <param name="preferredFPS" type="Integer" minvalue="0" maxvalue="2147483647" mandatory="false">
+            <description>Preferred frame rate per second. Mobile application should take this value into account for capturing and encoding video frame.</description>
+        </param>
+    </struct>
+```
+
+Like other parameters in VideoStreamingCapability, preferredFPS should be returned in GetSystemCapability RPC response.
+
+### iOS mobile proxy changes
+
+Extend SDLVideoStreamingCapability class to have preferredFPS property:
+
+SDLVideoStreamingCapability.h:
+
+```objc
+/**
+ *  Preferred frame rate (assuming the case where resolution is the same as preferredResolution)
+ *  Optional
+ */
+@property (nullable, strong, nonatomic) NSNumber<SDLInt> *preferredFPS;
+
+```
+
+SDLVideoStreamingCapability.m:
+```objc
+- (void)setPreferredFPS:(nullable NSNumber<SDLInt> *)preferredFPS {
+    [store sdl_setObject:preferredFPS forName:SDLRPCParameterNamePreferredFPS];
+}
+
+- (nullable NSNumber<SDLInt> *)preferredFPS {
+    return [store sdl_objectForName:SDLRPCParameterNamePreferredFPS ofClass:NSNumber.class error:nil];
+}
+```
+
+And then, modify SDLStreamingVideoLifecycleManager#didEnterStateVideoStreamStarting, so that preferredFPS value would be taken into account:
+```objc
+static NSUInteger const defaultFrameRate = 15;
+
+- (void)didEnterStateVideoStreamStarting {
+    ...
+    if (capability != nil) {
+        NSUInteger  preferredFPS = (capability.preferredFPS.integerValue == 0) ? defaultFrameRate : capability.preferredFPS.integerValue;
+        weakSelf.videoEncoderSettings[(__bridge NSString *)kVTCompressionPropertyKey_ExpectedFrameRate] = @(preferredFPS);
+        ...
+    } else {
+        ...
+        weakSelf.videoEncoderSettings[(__bridge NSString *)kVTCompressionPropertyKey_ExpectedFrameRate] = @(defaultFrameRate);
+    }
+
+}
+```
+Note that videoEncoderSettings property must be NSMutableDictionary as it now would be mutable.

```

Comment:
```suggestion
Note that `videoEncoderSettings` property must be NSMutableDictionary as it now would be mutable.
```

Reply:
Made NSMutableDictionary to coding phrase as well.